commands/init_test.go

```diff
@@ -1,32 +1,92 @@
 package commands
 
 import (
-	"github.com/github/hub/Godeps/_workspace/src/github.com/bmizerany/assert"
-	"github.com/github/hub/github"
+	"path/filepath"
+	"fmt"
 	"os"
-	"regexp"
 	"testing"
+
+	"github.com/github/hub/Godeps/_workspace/src/github.com/bmizerany/assert"
+	"github.com/github/hub/github"
 )
 
-func TestTransformInitArgs(t *testing.T) {
+func setup() {
```

Comment:
Oh `setup` will be shared across the whole package? That's no good, because it probably means that we're unable to create a `setup` method in another test file. Is there a way to have a setup block be defined only in the current file, for example with a func and a closure?

Also, our current setup edits an environment variable but doesn't reset it at the end of the test. Not a huge deal, but it would be ideal if we had a way of editing environment variables just for the duration of a single test and have it automatically reset, like Aruba does in our cukes.